src/main/java/com/iota/iri/network/impl/TransactionRequesterWorkerImpl.java

```diff
@@ -147,6 +147,9 @@ public void shutdown() {
      */
     private TransactionViewModel getTransactionToSendWithRequest() throws Exception {
         Hash tip = tipsViewModel.getRandomSolidTipHash();
+        if (tip == null) {
+            tip = tipsViewModel.getRandomNonSolidTipHash();
+        }
```

Comment:
Why did you create this change?
You think that due to local snapshots, tips that should be solid won't solidify?

I think usually we don't want to send "spam" tips that approve non-existent transactions.

Reply:
Fix: allow the transactionRequester to also send non-solid tips

Allow the transactionRequester to also send non-solid tips - otherwise we might have problems getting the tips solid (if we can not send any requests).

Reply:
@hmoog this should not make a difference, as `getRandomSolidTipHash` will call `getRandomNonSolidTipHash` if it has no tips
see: https://github.com/iotaledger/iri/issues/1229

Reply:
Oh, you are right it shouldn't make a difference since getRandomSolidTipHash() has an implementation that doesnt correctly reflects its name. Given the fact that we already have an issue to "fix" this behavior i would anyway keep this change in place to make it still work once getRandomSolidTipHash gets fixed.